File: Estrutura da Dados/Pilha/Linear/JAVA/pilha.java
import java.util.*;

class Pilha{

    private int topo;
    private int[] dados;

    public Pilha(){

        this.topo = -1;
        this.dados = new int[10];

    }


    //Verifica se pilha esta cheia 
    public boolean isFull(){

        if(topo == 10 - 1)
            return true;
        else
            return false;
    }

    //Verifica se pilha esta vazia
    public boolean isEmpty(){

        if(topo == -1)
            return true;
        else
            return false;
    }

    //Empilha 
    public void push(int valor){

        if(!isFull()){

            topo++;
            dados[topo] = valor;

        }
        else
            System.out.println("Pilha cheia\n");
    }

    public void pop(){

        if(!isEmpty()){

            System.out.println(dados[topo]);    //imprime valor q saiu da pilha 
            topo--;
        }
        else
            System.out.println("Pilha vazia\n");
    }





}

File: Estrutura da Dados/Pilha/Linear/C/pilha.c
#include "pilha.h"

//Cria nova Pilha 
Pilha* newPilha()  
{
    Pilha *p = (Pilha*)malloc(sizeof(Pilha));

    if(p == NULL)
        return NULL;

    //Atribui valores a Pilha 
    p->topo = -1;

}

//Verifica se pilha esta Cheia
bool isFull(Pilha *p)
{
    if(p->topo == MAX - 1)
        return true;
    else
        return false;

}

//Verifica se Pilha esta vazia
bool isEmpty(Pilha *p)
{
    if(p->topo == -1)
        return true;
    else
        return false;
}

//Empilha 
void push(Pilha *p,int valor)
{
    if(!isFull(p))
    {

        p->topo++;
        p->dados[p->topo] = valor;

    }
    else
        printf("Pilha Cheia!\n");

}

void pop(Pilha *p)
{
    if(!isEmpty(p))
    {   

        printf("%d",&p->dados[p->topo]);    //imprime Valor q saiu da Pilha
        p->topo--;

    }
    else
        printf("Pilha vazia!\n");


}

//Libera memória alocada por pilha 
void freePilha(Pilha *p)
{
    free(p);
}


